### DISCUSSION TITLE
DiscordChatExporter.osx-arm64.zip contains windows build instead

### DISCUSSION THREAD

**Discussion by redmorestudio** [Tyrrrz/DiscordChatExporter#1234] [General]
Just like the title says.

**Comment by redmorestudio** [comment]
and so does the other osx-x64 build too.

**Comment by Yudi** [comment]
There's no `.app` to open.

You must use the `Terminal.app`. Use Spotlight to search for it <kbd>⌘</kbd>+<kbd>Space</kbd>

Type `cd`, then hit <kbd>Space</kbd>, then drag and drop DCE's folder into the Terminal window, hit <kbd>Return</kbd>

If you've downloaded the ARM build, type the following:
```
dotnet DiscordChatExporter.dll
```

Then hit <kbd>Return</kbd>

Otherwise use:
```
./DiscordChatExporter
```

<img width="1226" alt="image" src="https://github.com/Tyrrrz/DiscordChatExporter/assets/12040397/d4900e37-641f-4a7a-8d8a-76f43a67436b">

**Reply by Tyrrrz** [reply]
`chmod +x` shouldn't be required anymore

**Reply by Yudi** [reply]
Yeah, it's no longer required

**Reply by Yudi** [reply]
One still needs to run `dotnet DiscordChatExporter.dll` instead of just `./DiscordChatExporter` because [macOS kills unsigned ARM executables on launch](https://eclecticlight.co/2021/01/26/when-you-dont-have-permission-to-run-an-app-on-an-m1-mac/).

The x64 build works fine through Rosetta.

**Comment by toadchild** [comment]
My mac doesn't have `dotnet` installed by default, and I didn't see that listed as a prerequisite anywhere.

**Reply by Tyrrrz** [reply]
It isn't a prerequisite for the app itself, but your Mac won't let you run an unsigned app, so you need to use the .NET CLI as a proxy. You can download it here: https://dotnet.microsoft.com/en-us/download

**Reply by toadchild** [reply]
I see. If it were actually assembled as a proper .app bundle it wouldn't be so difficult. I confirmed that if I go through all the .dylib files and right click -> open -> open on each of them in turn to bypass the app signing check, it allows me to run the tool via double-clicking on the DiscordChatExporter executable.

**Reply by Tyrrrz** [reply]
I created https://github.com/Tyrrrz/DiscordChatExporter/issues/1249

**Reply by prerakmody** [reply]
I can confirm that I had to use the same approach as @toadchild.
Right clicked on all the 16 .dylib files and opened them one-by-one. Finally got DiscordChatExporter to run then.

**Reply by laurikskl** [reply]
`sudo spctl --master-disable`

Should disable all protection and helps avoid having to click through those pesky "allow anyway" messages. 

After running the command you have to open Privacy & Security and there will be a new option there called "Always".

Remember to enable it again after you're done.
